plugins/catalog-backend/migrations/20230525141717_stitch_queue.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/**
+ * @param { import("knex").Knex } knex
+ */
+exports.up = async function up(knex) {
+  await knex.schema.alterTable('refresh_state', table => {
+    table
+      .dateTime('next_stitch_at')
+      .nullable()
+      .comment('Timestamp of when entity should be stitched');
+    table
+      .string('next_stitch_ticket')
+      .nullable()
+      .comment('Random value distinguishing stitch requests');
+    table.index('next_stitch_at', 'refresh_state_next_stitch_at_idx', {
+      predicate: knex.whereNotNull('next_stitch_at'),
+    });
+  });
+
+  // Look for things that are due for stitching, and move them over to the new
+  // system. An explanation on the length based ones: Before adding the new
+  // system, stitching was triggered by setting hashes to various hard coded
+  // strings - and we leverage the fact that those strings were always shorter
+  // than a real hash would be. There's also the case where we have not yet
+  // created the final_entities row corresponding to the refresh_state one. This
+  // is split into two statements because MySQL doesn't allow you to write to
+  // tables you're reading from.
+  const candidates = await knex
```

Comment:
this also worries me for taking a long time and causing deployments to fail... but I don't know of a better way.

On the other hand, refresh_state and final_entities are both much smaller than the search table, which is the root of my fears (previous experiences of big impact migration files)

Reply:
is there a way to seed the database with a few hundred thousand entities and just checking how long it'd take?

Reply:
Yeah good call - I think indeed that this is less risky than the search table updates in the past.

The performance tests have facilities for inserting massive amounts of data, see the synthetic load entities in https://github.com/backstage/backstage/blob/8d756968f9c8676c391868a5a9aae931994c877b/plugins/catalog-backend/src/tests/performance/stitchingPerformance.test.ts